packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -568,3 +569,143 @@ class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay
     return null;
   }
 }
+
+/// A gesture detector to detect concurrent gesture events for a text field.
+///
+/// Unlike a plain [GestureDetector] where a single gesture wins the gesture
+/// arena, text fields supports chains of concurrent events.
+///
+/// See also:
+///
+///  * [TextField], a Material text field which uses this gesture detector.
+///  * [CupertinoTextField], a Cupertino text field which uses this gesture
+///    detector.
+class TextSelectionGestureDetector extends StatefulWidget {
+  /// Create a [TextSelectionGestureDetector].
+  ///
+  /// Multiple callbacks can be called from the same sequence of gesture.
+  /// The [child] parameter must not be null.
+  const TextSelectionGestureDetector({
+    this.onTapDown,
+    this.onSingleTapUp,
+    this.onSingleTapCancel,
+    this.onSingleLongTapDown,
+    this.onDoubleTapDown,
+    @required this.child,
+  }) : assert(child != null);
+
+  /// Called each distinct time a tap was held momentarily or on a short tap.
+  final GestureTapDownCallback onTapDown;
+
+  /// Called for each distinct short tap. Also called on the first tap of a
+  /// double tap but not on the second tap.
+  final GestureTapUpCallback onSingleTapUp;
+
+  /// Called for each tap down that does not become a short tap such as a long
+  /// tap or drag.
+  final GestureTapCancelCallback onSingleTapCancel;
+
+  /// Called for a single long tap that's sustained but not necessarily lifted.
+  /// Not called for a double-tap-hold.
```

Comment:
I think this means: not called if it follows a "tap up" within kDoubleTapTimeout?

Reply:
We don't necessarily need to be too specific and redefine a double click in word. There's other little things like not being too far from the previous tap etc.